### PR TITLE
Added curvature and slope extraction routines

### PR DESCRIPTION
**Pull request by jeanbraun**
To @benbovy : I have added two routines called FastScape_Copy_Slope and FastScape_Copy_Curvature to extract slope and curvature from the current landscape. I have tested these routines in a range of situations and they seem to work. I have also added a few lines in the documentation.

### CODE REVIEW

**Comment by jeanbraun** [comment]
@benbovy are these failed checks important? Or can I continue with the merge pull?

File: src/Strati.f90 [hunk 1]
```diff
@@ -1,4 +1,4 @@
-subroutine Strati (h,b,F,nx,ny,xl,yl,reflector,nreflector,ireflector,istep,fields,nfield,vex,dt, &
```
Comment (benbovy) [inline src/Strati.f90]: Is this change somehow related to curvature and slope routines?

If not, it might be best to submit this change of ``Strati`` signature in another PR if you don't mind, so that it will be easier to track all changes when writing down the release notes for the next release. Or at least, when we'll merge this PR, add a few words about this change in the commit message.

File: src/Strati.f90 [hunk 2]
```diff
@@ -117,6 +117,44 @@ end subroutine slope
 
 !-------------------------------------------------------------------
 
+subroutine curvature (h,curv,nx,ny,dx,dy)
```
Comment (benbovy) [inline src/Strati.f90]: Wouldn't be better to put this routine in a separate source file, e.g., `TerrainDerivatives.f90`, as this is not really related to stratigraphy?

File: src/Strati.f90 (hunk 1 above)
Comment (jeanbraun) [reply src/Strati.f90]: This change is related to the passing of an array (h) that is not needed in the routine. It is a very minor point but I'm happy to put it in another PR.

File: src/Strati.f90 (hunk 2 above)
Comment (jeanbraun) [reply src/Strati.f90]: I thought about it too and will do so. Thanks

**Comment by jeanbraun** [comment]
@benbovy Note that I have added a new source file (TerrainDerivatives.f90) as suggested that also required to change the CMakeLists file. I have also slightly reformatted some of the source files (using the Fortran template) which has led to some changes related to indentation. I hope that it does not make the merge too bulky. Should I merge?

**Comment by jeanbraun** [comment]
Hi @bbovy, I have merged the branch with the curvature and slope (and the new routine) onto master. Could you please try to put it on condo so that Lena can install it on her computer. Many thanks.
